exercise_2/main.cpp

```diff
@@ -0,0 +1,85 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+#include "Exam.h"
+#include "Student.h"
+#include "Teacher.h"
+#include "Course.h"
+#include <string>
+
+
+void addExamToMap(std::istringstream &iss_record, std::map<int,Exam> &exams) {
```

Comment:
Please note, that all of the adding entries to map methods are more or less the very same.
They differ in a container being used to keep entires and they differ in the way how data is read from the stream.

Could you please try to make it using somehow common code?